libgadget/fofpetaio.c

```diff
@@ -127,8 +127,10 @@ struct PartIndex {
 };
 
 static int fof_sorted_layout(int i, const void * userdata) {
-    return P[i].targettask;
+    int * targetTask = (int *) userdata;
```

Comment:
Was it important to carry targetTask with P in exchange? I think not. Did you check this?

Reply:
I checked the output was the same and it doesn't seem important from looking at the code. Carrying origintask is important.